### DISCUSSION TITLE
Is anyone considering making a version outside of React? I desire a version with Web Components and I can help!

### DISCUSSION THREAD

**Discussion by erkylima** [xyflow/xyflow#4232] [Ideas]
I've been looking for a solution as good for Node Base UI as React Flow for Web Components with Vanilla JavaScript. In this relentless search, I've come to the conclusion that we definitely need a vanilla JavaScript version for React Flow. This will surely be an excellent contribution to the community, and it would be an honor to help! I personally really need this.

**Comment by oxdev03** [comment]
Did you already check here?
https://github.com/xyflow/awesome-node-based-uis

**Comment by lublak** [comment]
There could be a option with svelte: https://svelte.dev/docs/custom-elements-api @erkylima you could try it out.

**Comment by alexanderhorner** [comment]
Curious what's the argument against using any front end library? Why not use react or svelte?

**Reply by samuelstroschein** [reply]
Distributing a component as a library.

I'm interested in using React Flow to write a component that developers can use in any framework like https://github.com/nicoespeon/gitgraph.js/. Web Component is a hard requirement. The component should be framework-independent. React, Svelte, Vue, Angular, whatever devs should be able to use the component. React is at odds with that requirement.

**Reply by lublak** [reply]
@alexanderhorner because vanilla is still a thing? "Why not use react or svelte?" The question is also: Why not vanilla? or web components. Every developer has different requirements. Whether the library offers the request is up to the maintainer.

**Reply by alexanderhorner** [reply]
@lublak Yes, it's up to the maintainers of course.

Im just curious. There's a reason why people prefer react/svelte/etc over vanilla for these kinds of things. If you're bringing in a library for drawing graphs, why wouldn't you also bring a library for the user interface in general?

**Comment by sporritt** [comment]
You could try https://jsplumbtoolkit.com

We'd be well up for making a web components version, it's something we've been thinking about for a while now

**Comment by JulianCataldo** [comment]
If you're still searching, I've just released this PoC, a web components kit for node based UIs:

https://github.com/JulianCataldo/node-flow-elements

![](https://private-user-images.githubusercontent.com/603498/421967413-aed258f5-d2d7-4e4e-ad9a-6a70b47f68dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM5MzQwNjAsIm5iZiI6MTc0MzkzMzc2MCwicGF0aCI6Ii82MDM0OTgvNDIxOTY3NDEzLWFlZDI1OGY1LWQyZDctNGU0ZS1hZDlhLTZhNzBiNDdmNjhkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwNlQxMDAyNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02Y2I3ODJjN2FlNGVhNGY3NzJkMTA2NzAzMmE4Zjg0ZGYyNGRjZTE4ZDljMzNiZjRlMTA5ZTliZGQzMzc5MDdkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Hpa5e_-075FlsdB5q0A_k3esXBxEA1s4swOGEdAVQmM)

It uses TC 39 signals proposal and is compatible with any UI framework of your choice.

I took a lot of inspiration from xyflow ;)
